compiler/hash-pest-parser/src/translate.rs

```diff
@@ -907,9 +907,25 @@ where
                 }
             }
             Rule::compound_pattern => {
-                let pats = ab.try_collect(pair.into_inner().map(|p| self.transform_pattern(p)))?;
-
-                Ok(ab.node(Pattern::Or(OrPattern { variants: pats })))
+                let mut pairs = pair.into_inner();
+                let first = self.transform_pattern(pairs.next().unwrap())?;
+
+                // if there is only one pattern within the compound pattern, we don't want to always wrap it an 'Or'
+                // variant since this is redundant
+                match pairs.next() {
+                    None => Ok(first),
+                    Some(pat) => {
+                        let second = self.transform_pattern(pat)?;
+
+                        // collect any remaining patterns in the or secquence
+                        let variants = vec![first, second]
+                            .into_iter()
+                            .chain(ab.try_collect(pairs.map(|p| self.transform_pattern(p)))?)
```

Comment:
Do you need to `try_collect` here?

Reply:
Yes, since the patterns after the first and second are still to be transformed and there may be many patterns. I think the best way is to use a `try_collect`, unless there is a better way to unwrap a `Vec` of `ParseResult` into `AstNode<Pattern>` with the `chain` call?